src/de/willuhn/jameica/fibu/server/KontoImpl.java

```diff
@@ -226,14 +226,13 @@ private double getUmsatz(DBIterator buchungen, DBIterator hilfsbuchungen) throws
 
     int kontoArt = getKontoArt().getKontoArt();
     Kontotyp typ = getKontoTyp();
+    //Split-Hauptbuchungen rausfiltern
+    buchungen.addFilter("NOT EXISTS(SELECT 1 FROM buchung b WHERE b.split_id = buchung.id)");
```

Comment:
Hier wird der DBIterator von außen als Parameter reingereicht. Daher sollten hier keine weiteren SQL-WHERE-Bedingungen hinzugefügrt werden. Das Risiko ist zu hoch, dass das irgend Nebeneffekte hat.

Da aber ohnehin nur server-seitig über die Buchungen iteriert wird, ohne einzelne Buchungen an die UI zu übertragen, kann man die unerwünschten Buchungen hier auch einfach überspringen.

Wenn ich das richtig verstanden habe, sollte ja der Betrag der Hauptbuchung identisch mit der Summe der Split-Buchungen sein, oder? In dem Fall könnte man hier ja umgekehrt die Splitbuchungen weglassen und nur die Hauptbuchungen addieren. Etwa so:

```java
    while (buchungen.hasNext())
    {
      Transfer t = (Transfer) buchungen.next();

      // "getSplitID()" müsste man ggf. noch hinzufügen.
      // Das sollte doch dann die Split-Buchungen ignorieren.
      if ((t instanceof Buchung) && ((Buchung)t).getSplitID() != null)
        continue;

      if (t.getSollKonto().equals(this))
        soll += t.getBetrag();
      else
        haben += t.getBetrag();
    }
```

Reply:
Stimmt. Das mit der Einschränkung auf das Konto ist natürlich korrekt. Da macht das hier Sinn.